util.go

```diff
@@ -0,0 +1,159 @@
+package graphql
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+const TAG = "json"
+
+// can't take recursive slice type
+// e.g
+// type Person struct{
+//	Friends []Person
+// }
+// it will throw panic stack-overflow
+func BindFields(obj interface{}) Fields {
+	v := reflect.ValueOf(obj)
+	fields := make(map[string]*Field)
+
+	for i := 0; i < v.NumField(); i++ {
+		typeField := v.Type().Field(i)
+
+		tag := typeField.Tag.Get(TAG)
+		if tag == "-" {
+			continue
+		}
+		var graphType Output
+		if typeField.Type.Kind() == reflect.Struct {
+
+			structFields := BindFields(v.Field(i).Interface())
+			if tag == "" {
+				fields = appendFields(fields, structFields)
+				continue
+			} else {
+				graphType = NewObject(ObjectConfig{
+					Name:   tag,
+					Fields: structFields,
+				})
+			}
+		}
+
+		if tag == "" {
+			continue
+		}
+
+		if graphType == nil {
+			graphType = getGraphType(typeField.Type)
+		}
+		fields[tag] = &Field{
+			Type: graphType,
+			Resolve: func(p ResolveParams) (interface{}, error) {
+				return extractValue(tag, p.Source), nil
+			},
+		}
+	}
+	return fields
+}
+
+func getGraphType(tipe reflect.Type) Output {
+	kind := tipe.Kind()
+	switch kind {
+	case reflect.String:
+		return String
+	case reflect.Int:
+		return Int
+	case reflect.Float32:
+	case reflect.Float64:
+		return Float
+	case reflect.Bool:
+		return Boolean
+	case reflect.Slice:
+		return getGraphList(tipe)
+	}
+	return String
+}
+
+func getGraphList(tipe reflect.Type) *List {
+	switch tipe {
+	case reflect.TypeOf([]int{}):
```

Comment:
Not a fan of all the `reflect` usage here, you've already got a `reflect.Type`, you can use it. like here `case tipe.Kind() == reflect.Slice && tipe.Elem().Kind() == reflect.Int` and other tests `case tipe.Kind() == reflect.Slice && tipe.Elem().Kind() == reflect.Int8`, etc...